FantasyUniverse/Networking/Cache.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Cache.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 19.04.2022.
+//
+
+import Foundation
+
+protocol Cachable {
+    associatedtype Element
+    associatedtype Searched
+        
+    func getElements() -> [Element]
+    func cacheElements(_ objects: [Element])
+    func addToCache(_ objects: [Element])
+    func cleanCache()
+    func cacheIsEmpty() -> Bool
+    func cacheContainElement(searched: Searched) -> Bool
+    
+}
+
+class CachedQuestions: Cachable {
+
+    private var cachedElements: [QuestionCollection] = []
+    
+    func getElements() -> [QuestionCollection] {
+        return cachedElements
+    }
+    
+    func cacheElements(_ elements: [QuestionCollection]) {
+        cachedElements = elements
+    }
+    
+    func addToCache(_ elements: [QuestionCollection]) {
+        cachedElements += elements
+    }
+
+    func cleanCache() {
+        cachedElements = []
+    }
+   
+    func cacheIsEmpty() -> Bool {
+        return cachedElements.isEmpty
+    }
+    
+    func cacheContainElement(searched: QuestionSet) -> Bool {
```

Comment:
I'd make it similar to the `contains` method from the array type so the `Searchable` associated type and this intricate knowledge of model can be abstracted outside of cache.